src/gui/elevation/qgselevationprofilecanvas.cpp

```diff
@@ -844,7 +845,10 @@ void QgsElevationProfileCanvas::refresh()
 
   const QList< QgsMapLayer * > layersToGenerate = layers();
   QList< QgsAbstractProfileSource * > sources;
-  sources.reserve( layersToGenerate .size() );
+  QList< QgsAbstractProfileSource * > registrySources = QgsApplication::profileSourceRegistry()->profileSources();
```

Comment:
```suggestion
  const QList< QgsAbstractProfileSource * > registrySources = QgsApplication::profileSourceRegistry()->profileSources();
```

Reply:
Addressed in https://github.com/qgis/QGIS/pull/57311/commits/de9abdfeaa806000a060d22b308048dd245beb75